bundles/framework/mydata/service/MyDataService.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * @class Oskari.mapframework.bundle.mydata.service.MyDataService
+ *
+ * Requests for a search to be made with the given query and provides
+ * callbacks
+ */
+ Oskari.clazz.define('Oskari.mapframework.bundle.personaldata.service.MyDataService',
+
+ /**
+  * @method create called automatically on construction
+  * @static
+  */
+
+ function (uiHandler, updater) {
+    this._uiHandler = uiHandler;
+    this._updater = updater;
+ }, {
+    /** @static @property __qname fully qualified name for service */
+    __qname: 'Oskari.mapframework.bundle.personaldata.service.MyDataService',
+    /**
+     * @method getQName
+     * @return {String} fully qualified name for service
+     */
+    getQName: function () {
+        return this.__qname;
+    },
+    /** @static @property __name service name */
+    __name: 'MyDataService',
+    /**
+     * @method getName
+     * @return {String} service name
+     */
+    getName: function () {
+        return this.__name;
+    },
+    addTab: function (id, title, component, handler) {
+        handler.addStateListener(this._updater);
```

Comment:
We could remove the updater function and just propagate the event to notify uiHandler statelisteners, triggering an update call that is defined for uiHandler state changes:
```suggestion
        handler.addStateListener(this._uiHandler.notify);
```